yb-voyager/cmd/importData.go

```diff
@@ -933,12 +933,19 @@ func executeSqlFile(file string) {
 
 	var errOccured = 0
 	sqlStrArray := createSqlStrArray(file, "")
-	for _, sqlStr := range sqlStrArray {
+	retryCount := INDEX_RETRY_COUNT
+	for i := 0; i < len(sqlStrArray); {
+		sqlStr := sqlStrArray[i]
 		log.Infof("Run query %q on target %q", sqlStr[1], target.Host)
 		_, err := conn.Exec(context.Background(), sqlStr[0])
 		if err != nil {
-			log.Errorf("Run query %q on target %q: %s", sqlStr[1], target.Host, err)
-			if strings.Contains(err.Error(), "already exists") {
+			log.Errorf("Index creation failed %s", err)
+			if strings.Contains(strings.ToLower(err.Error()), strings.ToLower(SCHEMA_VERSION_MISMATCH_ERR)) &&
+				retryCount > 0 {
```

Comment:
Don't update the counters in reverse direction. They are unnecessarily complicated for understanding. Inside the suggested `execSqlStatementWithRetries(conn, sql)` function have a straight-forward loop on retryCount (i.e. `for retryCount := 1; retryCount <= DDL_STATEMENT_MAX_RETRY_COUNT; retryCount++ {`) .